### ISSUE TITLE
UX - copy and paste: paste should paste contents at mouse cursor

### ISSUE THREAD

**Issue by muziker** [pure-data/pure-data#864]
Copying and pasting currently pastes the contents into an offset of the previous position. Pasting should paste the contents at the current mouse position, just like how the hotkeys ctrl-1/2/3 pastes the objects at the mouse position

**Comment by umlaeute** [comment]
i don't know.

i guess [citation needed] people will be upset if copy-paste won't reproduce the exact information, including (absolute, not only relative) positions of objects.

**Comment by jmmmp** [comment]
I'm used to the absolute position also be pasted, and it helps a lot with my own programming. For me it would be a disadvantage if this would be lost.

**Comment by porres** [comment]
I usually copy one thing from one patch to another and I expect it to be placed in the same position from the first one. That is really helpful for me when doing documentation (and I have written almost 600 help files).

**Comment by Lucarda** [comment]
I agree with @jmmmp and @porres. I find very useful to paste in the same coords as the source.

**Comment by muziker** [comment]
Why not have a ctrl+SHIFT+v to paste at mouse location? 

When copying from within the same patch, pasting places these objects close to where the original is, but usually the object needs to be pasted into an area of the canvas that is elsewhere.

**Comment by umlaeute** [comment]
most likely because this creates a vertical slider right now...

**Comment by jmmmp** [comment]
that could be a good variant.
since several versions, pasting places the objects exactly where the
original was (before it was 10 pixels SE).
I would say that for me, I need copy-pasting often rather to place objects
exactly as the original - but that might not reflect the experience of many
users.
As long as both options are available, I would be happy with whichever
shortcut is used for what.

Am Mi., 22. Jan. 2020 um 05:34 Uhr schrieb muziker <notifications@github.com

**Comment by muziker** [comment]
> most likely because this creates a vertical slider right now...

Well that's unfortunate ... But yes i think there's more copy and pasting than creating vert sliders

**Comment by umlaeute** [comment]
for (copy-and-)paste we already have the universal <kbd>Ctrl</kbd>+<kbd>v</kbd>.

do you think that there's enough pasting to justify two shortcuts and take one away from the vertical sliders?

**Comment by muziker** [comment]
I don't have access to the statistics, so i cannot answer that question, but maybe a soft switch in configuration file would be work?

**Comment by umlaeute** [comment]
probably more interesting would be a feature that allows the current selection to be moved to the current pointer position.

something like:
- <kbd>Ctrl</kbd>+<kbd>v</kbd> to copy the current selection to the original position
- <kbd>Home</kbd> to move the selection towards the mouse (so that the upper-right corner is under the pointer position)
- the selection sticks to the pointer until you put it down with a mouse-click.

the <kbd>Home</kbd> key is entirely arbitrary. check what other software is using (and complain that it is on the wrong side of the keyboard)
